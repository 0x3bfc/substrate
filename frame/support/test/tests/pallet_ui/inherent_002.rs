#[frame_support::pallet]
mod pallet {
	use frame_support::pallet_prelude::{ModuleInterface, ProvideInherent};
	use frame_system::pallet_prelude::BlockNumberFor;

	#[pallet::config]
	pub trait Trait: frame_system::Trait {}

	#[pallet::module]
	pub struct Module<T>(core::marker::PhantomData<T>);

	#[pallet::module_interface]
	impl<T: Trait> ModuleInterface<BlockNumberFor<T>> for Module<T> {}

	#[pallet::call]
	impl<T: Trait> Module<T> {}

	#[pallet::inherent]
	impl<T: Trait> ProvideInherent for Module<T> {}
}

fn main() {
}
