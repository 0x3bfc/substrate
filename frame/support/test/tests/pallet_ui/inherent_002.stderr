error[E0046]: not all trait items implemented, missing: `Call`, `Error`, `INHERENT_IDENTIFIER`, `create_inherent`
  --> $DIR/inherent_002.rs:19:2
   |
19 |     impl<T: Trait> ProvideInherent for Module<T> {}
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ missing `Call`, `Error`, `INHERENT_IDENTIFIER`, `create_inherent` in implementation
   |
   = help: implement the missing item: `type Call = Type;`
   = help: implement the missing item: `type Error = Type;`
   = help: implement the missing item: `const INHERENT_IDENTIFIER: [u8; 8] = value;`
   = help: implement the missing item: `fn create_inherent(_: &pallet::sp_api_hidden_includes_pallet::hidden_include::inherent::InherentData) -> std::option::Option<<Self as pallet::sp_api_hidden_includes_pallet::hidden_include::inherent::ProvideInherent>::Call> { todo!() }`
